apps/server/test/lexical/server/code_intelligence/completion/env_test.exs

```diff
@@ -228,12 +233,132 @@ defmodule Lexical.Server.CodeIntelligence.Completion.EnvTest do
     end
   end
 
+  describe "in_context?(env, :struct_arguments)" do
+    def my_module(text) do
+      """
+      defmodule MyModule do
+        #{text}
+      end
+      """
+    end
+
+    def func_variable(text) do
+      """
+      def func do
+        #{text}
+      end
+      """
+    end
+
+    def func_args(text) do
+      """
+      def func(#{text}) do
+      end
+      """
+    end
+
+    test "is true if the cursor after curly" do
+      env = "%User{|}" |> my_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is true when the struct is in the function variable" do
+      env = "%User{|}" |> func_variable() |> my_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is true when the struct is in the function arguments" do
+      env = "%User{|}" |> func_args() |> my_module() |> new_env()
+      assert in_context?(env, :struct_arguments)
+    end
+
+    test "is false when the cursor is not in a container" do
+      env = new_env("%User{|}")
```

Comment:
this looks like it's in a struct reference to me.

(This is one reason I find `container_cursor_to_quoted` annoying, it seems to make decisions for you that I don't understand)

Reply:
I have no better idea.